books/RayTracingInOneWeekend.html

```diff
@@ -502,54 +502,103 @@
-virtual viewport through which to pass our scene rays. For the standard pixel spacing (equally
-spaced in both the horizontal and vertical direction), the viewport's aspect ratio should be the
-same as our rendered image. We'll just pick a viewport two units in height. We'll also set the
-distance between the projection plane and the projection point to be one unit. This is referred to
-as the _focal length_, not to be confused with the _focus distance_, which we'll present later.
+  virtual _viewport_ through which to pass our scene rays.
+The viewport is a virtual rectangle in the 3D world that contains the grid of image pixel locations.
+If pixels are spaced the same distance horizontally as they are vertically, the viewport that
+  bounds them will have the same aspect ratio as the rendered image.
+The distance between two adjacent pixels is called the pixel spacing, and square pixels is the
+  standard.
+
+To start things off, we'll choose an arbitrary viewport height of 2.0, and scale the viewport width
+  to give us the desired aspect ratio.
+Here's a snippet of what this code will look like:
+
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
+    auto aspect_ratio = 16.0 / 9.0;
+    int image_width = 400;
+
+    // Calculate the image height, and ensure that it's at least 1.
+    int image_height = static_cast<int>(image_width / aspect_ratio);
+    image_height = (image_height < 1) ? 1 : image_height;
+
+    // Calculate the viewport height. Since it's real valued, we don't check for values less then 1.
+    auto viewport_height = 2.0;
+    auto viewport_width = viewport_height * (static_cast<double>(image_width)/image_height);
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    [Listing [image-setup]: Rendered image setup]
+
+(If you're wondering why we don't just use `aspect_ratio` when computing `viewport_width`, it's
+because the _actual_ aspect ratio is the integer-based ratio of image width over image height, which
+will yield zero for values less than one.)
 
-I’ll put the “eye” (or camera center if you think of a camera) at $(0,0,0)$.
+Next we will define the camera center: a point in 3D space from which all scene rays will originate
+  (this is also commonly referred to as the _eye point_).
+The vector from the camera center to the viewport center will be orthogonal to the viewport.
+We'll initially set the distance between the viewport and the camera center point to be one unit.
+This distance is often referred to as the _focal length_.
+
+I’ll put the camera center at $(0,0,0)$.
 I will have the y-axis go up, and the x-axis to the right.
-In order to respect the convention of a right handed coordinate system, into the screen is the
-negative z-axis.
+Since we use a right-handed coordinate system, the vector into the screen is the negative z-axis.
```

Comment:
Couple of things

1. I don't think that we need to explain why we're doing y up versus z up, but it's something to consider explaining
2. This is the first time we're mentioned that we're using a right-handed coordinate system. There was no previous mention of what that means, nor was there a previous mention that we're doing this. This reads kind of confusingly. "Since we are doing X"... wait we are? Since when. You need to rephrase this to state that we are in right-handed coordinate systems, and we should also probably say the choice is an arbitrary one

Reply:
I would stay far away from the Y-up/Z-up debate. Just tell them what our convention is and move along.

Will reword with respect to RH coordinates.

Reply:
Just reworded to:
> I will have the y-axis go up, the x-axis to the right, and the negative z-axis pointing in the
> viewing direction. (This is commonly referred to as _right-handed coordinates_.)

Reply:
Latest:
> For simplicity we'll start with the camera center at $(0,0,0)$.
> We'll also have the y-axis go up, the x-axis to the right, and the negative z-axis pointing in the
> viewing direction. (This is commonly referred to as _right-handed coordinates_.)